### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by sithlord48**
merge fixes for #11 #9 #8 #7

### CODE REVIEW

File: widgets/BGDialog.cpp [hunk 1]
```diff
@@ -21,20 +21,21 @@ BGDialog::BGDialog(QWidget *parent) :
 	QDialog(parent, Qt::Tool), field(0)
 {
 	setWindowTitle(tr("Décor"));
+	imageFrame = new QFrame();
```
Comment (myst6re) [inline widgets/BGDialog.cpp]: Good idea

File: widgets/BGDialog.cpp [hunk 2]
```diff
@@ -21,20 +21,21 @@ BGDialog::BGDialog(QWidget *parent) :
...
 	pal.setColor(QPalette::Inactive, QPalette::Window, Qt::black);
 	pal.setColor(QPalette::Disabled, QPalette::Window, .60*Qt::black);
-	scrollArea->setPalette(pal);
+	imageFrame->setPalette(pal);
+
 
 	image = new ApercuBGLabel();
 	image->setAlignment(Qt::AlignCenter);
-	scrollArea->setWidget(image);
+	QVBoxLayout * imageLayout = new QVBoxLayout();
```
Comment (myst6re) [inline widgets/BGDialog.cpp]: Please write pointers declaration with one space before and no space after: Foo *bar;
The coding style in MR need changes, but for now I prefer to keep everything consistant.

File: widgets/BGDialog.cpp [hunk 3]
```diff
@@ -234,5 +235,9 @@ void BGDialog::changeZ(int value)
 void BGDialog::updateBG()
 {
 	if(!field)	return;
-	image->setPixmap(QPixmap::fromImage(field->background()->openBackground(params, z, layers)));
+	image->setPixmap(QPixmap::fromImage(field->background()->openBackground(params, z, layers)).scaled(image->width(),image->height(),Qt::KeepAspectRatio,Qt::SmoothTransformation));
```
Comment (myst6re) [inline widgets/BGDialog.cpp]: Do you really need to scale the background? I need to check if SmoothTransformation is the best option.

File: widgets/BGDialog.cpp [hunk 4]
```diff
@@ -234,5 +235,9 @@ void BGDialog::changeZ(int value)
 void BGDialog::updateBG()
 {
 	if(!field)	return;
-	image->setPixmap(QPixmap::fromImage(field->background()->openBackground(params, z, layers)));
+	image->setPixmap(QPixmap::fromImage(field->background()->openBackground(params, z, layers)).scaled(image->width(),image->height(),Qt::KeepAspectRatio,Qt::SmoothTransformation));
+}
+void BGDialog::resizeEvent(QResizeEvent *event)
+{
+	if(event->type()==QEvent::Resize){updateBG();}
```
Comment (myst6re) [inline widgets/BGDialog.cpp]: Please use spaces and line breaks

File: widgets/BGDialog.h [hunk 5]
```diff
@@ -49,6 +49,9 @@ private slots:
 	QHash<quint8, quint8> params;
 	bool layers[4];
 	qint16 x[3], y[3], z[3];
+	QFrame *imageFrame;
+protected:
+ void resizeEvent(QResizeEvent*);
```
Comment (myst6re) [inline widgets/BGDialog.h]: need a tabulation here

File: widgets/ModelManager.cpp [hunk 6]
```diff
@@ -22,7 +22,7 @@
 ModelManager::ModelManager(const QGLWidget *shareWidget, QWidget *parent) :
 	QDialog(parent, Qt::Tool), _field(0), fieldModelLoader(0)
 {
-	qreal scale = qApp->desktop()->logicalDpiX()/96;
+	//qreal scale = qApp->desktop()->logicalDpiX()/96;
```
Comment (myst6re) [inline widgets/ModelManager.cpp]: Remove this?

File: widgets/WalkmeshManager.cpp [hunk 7]
```diff
@@ -437,7 +436,7 @@ QWidget *WalkmeshManager::buildCameraRangePage()
 	layout->addWidget(group1);
 	layout->addWidget(group2);
 	layout->addWidget(group3);
-	layout->addStretch();
+	//layout->addStretch();
```
Comment (myst6re) [inline widgets/WalkmeshManager.cpp]: I need to check if addStretch can be remove here.

File: widgets/BGDialog.cpp (hunk 3 above)
Comment (sithlord48) [inline widgets/BGDialog.cpp]: scaling needs to happen or the image will be the size in game (320x240).

File: widgets/WalkmeshManager.cpp (hunk 7 above)
Comment (sithlord48) [inline widgets/WalkmeshManager.cpp]: it seams to do nothing. i have left it commented so others can try with / with out it to see a differance.

**Pull request merged**